src/horust/healthcheck.rs

```diff
@@ -133,13 +140,71 @@ pub fn prepare_service(healthiness: &Healthiness) -> Result<(), std::io::Error>
     Ok(())
 }
 
+fn client_mock() -> bool {
+    loop {
+        match TcpStream::connect("127.0.0.1:9999") {
+            Ok(mut _stream) => {
+                println!("Successfully connected to server in port 9999");
+                return true;
+            }
+            Err(e) if e.kind() == ErrorKind::ConnectionRefused => {
+                continue;
+            }
+            Err(e) => {
+                println!("Failed to connect: {}", e);
+                return false;
+            }
+        }
+    }
+}
+
+fn start_server() {
+    let listener = TcpListener::bind("127.0.0.1:9999").unwrap();
+    listener
+        .set_nonblocking(true)
+        .expect("Cannot set to non-blocking");
+    // accept connections and process them, spawning a new thread for each one
+    println!("Server listening on port 9999");
+    for stream in listener.incoming() {
+        match stream {
+            Ok(stream) => {
+                // connection succeeded
+                handle_test_connection(stream);
+            }
+            Err(e) if e.kind() == ErrorKind::WouldBlock => {
+                continue;
+                /* connection failed */
+            }
+            Err(_) => {}
+        }
+    }
+    // close the socket server
+    drop(listener);
+}
+
+fn handle_test_connection(mut stream: TcpStream) {
```

Comment:
you should move test related functions iniside the #mod test (line 197) otherwise you will get these nice "Unusued" warnings.